### ISSUE TITLE
Does it work with CKAN 2.6 and Apache WSGI ? 

### ISSUE THREAD

**Issue by gsueur** [NaturalHistoryMuseum/ckanext-ldap#9]
Hi,

I've been trying to install and make the ldap plugin work for a while without success until now. It has been installed manually, with setup.py develop
The funny thing is when I run CKAN with paster serve, it loads correctly the ldap login template but crashes on the form submission : 
`File '/usr/lib/ckan/default/local/lib/python2.7/site-packages/ldap/ldapobject.py', line 77 in __init__
  self._l = ldap.functions._ldap_function_call(ldap._ldap_module_lock,_ldap.initialize,uri)
File '/usr/lib/ckan/default/local/lib/python2.7/site-packages/ldap/functions.py', line 66 in _ldap_function_call
  result = func(*args,**kwargs)
LDAPError: (2, 'No such file or directory')
`

When I try to use it with Apache and WSGI module, which is my production environment, it first doesn't use the correct template (but I replace the standard one with the ldap one) and then crashes on the login page load without very obvious reason. I drop the full WSGI error stack below.
Thanks for any clue you may give us. 

Best regards

Guillaume 

[Wed Aug 24 09:49:45.850605 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581] WSGI Variables
[Wed Aug 24 09:49:45.850610 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581] --------------
[Wed Aug 24 09:49:45.850614 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   apache.version: (2, 4, 10)
[Wed Aug 24 09:49:45.850619 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   application: <fanstatic.publisher.Delegator object at 0x7fd3c5f51f50>
[Wed Aug 24 09:49:45.850624 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   beaker.cache: <beaker.cache.CacheManager object at 0x7fd3c5f3f990>
[Wed Aug 24 09:49:45.850630 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   beaker.get_session: <bound method SessionMiddleware._get_session of <beaker.middleware.SessionMiddleware object at 0x7fd3c5f3f3d0>>
[Wed Aug 24 09:49:45.850635 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   beaker.session: {'_accessed_time': 1472024985.837062, '_creation_time': 1472024985.837062}
[Wed Aug 24 09:49:45.850641 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   fanstatic.needed: <fanstatic.core.NeededResources object at 0x7fd3c4439290>
[Wed Aug 24 09:49:45.850646 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   mod_wsgi.application_group: 'ckan.aua-toulouse.org|'
[Wed Aug 24 09:49:45.850669 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   mod_wsgi.callable_object: 'application'
[Wed Aug 24 09:49:45.850682 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   mod_wsgi.daemon_connects: '1'
[Wed Aug 24 09:49:45.850687 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   mod_wsgi.daemon_restarts: '0'
[Wed Aug 24 09:49:45.850692 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   mod_wsgi.daemon_start: '1472024985827993'
[Wed Aug 24 09:49:45.850696 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   mod_wsgi.enable_sendfile: '0'
[Wed Aug 24 09:49:45.850701 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   mod_wsgi.handler_script: ''
[Wed Aug 24 09:49:45.850706 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   mod_wsgi.input_chunked: '0'
[Wed Aug 24 09:49:45.850711 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   mod_wsgi.listener_host: ''
[Wed Aug 24 09:49:45.850716 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   mod_wsgi.listener_port: '80'
[Wed Aug 24 09:49:45.850721 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   mod_wsgi.process_group: 'ckan_default'
[Wed Aug 24 09:49:45.850726 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   mod_wsgi.queue_start: '1472024985827599'
[Wed Aug 24 09:49:45.850730 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   mod_wsgi.request_handler: 'wsgi-script'
[Wed Aug 24 09:49:45.850735 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   mod_wsgi.request_start: '1472024985827337'
[Wed Aug 24 09:49:45.850740 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   mod_wsgi.script_reloading: '1'
[Wed Aug 24 09:49:45.850745 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   mod_wsgi.script_start: '1472024985828049'
[Wed Aug 24 09:49:45.850750 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   mod_wsgi.version: (4, 3, 0)
[Wed Aug 24 09:49:45.850755 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   paste.cookies: (<SimpleCookie: >, '')
[Wed Aug 24 09:49:45.850760 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   paste.registry: <paste.registry.Registry object at 0x7fd3c44abe10>
[Wed Aug 24 09:49:45.850765 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   paste.throw_errors: True
[Wed Aug 24 09:49:45.850771 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   pylons.action_method: <bound method UserController.login of <ckan.controllers.user.UserController object at 0x7fd3c44396d0>>
[Wed Aug 24 09:49:45.850777 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   pylons.controller: <ckan.controllers.user.UserController object at 0x7fd3c44396d0>
[Wed Aug 24 09:49:45.850782 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   pylons.environ_config: {'session': 'beaker.session', 'cache': 'beaker.cache'}
[Wed Aug 24 09:49:45.850787 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   pylons.pylons: <pylons.util.PylonsContext object at 0x7fd3c4439610>
[Wed Aug 24 09:49:45.850792 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   pylons.routes_dict: {'action': u'login', 'controller': u'user'}
[Wed Aug 24 09:49:45.850797 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   repoze.who.api: <repoze.who.api.API object at 0x7fd3c44ab8d0>
[Wed Aug 24 09:49:45.850802 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   repoze.who.logger: <logging.Logger object at 0x7fd3c5f5dfd0>
[Wed Aug 24 09:49:45.850807 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   repoze.who.logins: 0
[Wed Aug 24 09:49:45.850813 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   repoze.who.plugins: {'ckan.lib.authenticator:UsernamePasswordAuthenticator': <ckan.lib.authenticator.UsernamePasswordAuthenticator object at 0x7fd3c5f79710>, 'friendlyform': <FriendlyFormPlugin 140547535208848>, 'auth_tkt': <CkanAuthTktCookiePlugin 140547535208784>}
[Wed Aug 24 09:49:45.850825 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   routes.route: <routes.route.Route object at 0x7fd3c5f12650>
[Wed Aug 24 09:49:45.850830 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   routes.url: <routes.util.URLGenerator object at 0x7fd3c44394d0>
[Wed Aug 24 09:49:45.850835 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   webob._parsed_query_vars: (GET([]), '')
[Wed Aug 24 09:49:45.850840 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   webob.adhoc_attrs: {'response': <Response at 0x7fd3c4439490 200 OK>, 'language': 'en-us'}
[Wed Aug 24 09:49:45.850845 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   wsgi process: 'Multi process AND threads (?)'
[Wed Aug 24 09:49:45.850850 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   wsgi.file_wrapper: <type 'mod_wsgi.FileWrapper'>
[Wed Aug 24 09:49:45.850856 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581]   wsgiorg.routing_args: (<routes.util.URLGenerator object at 0x7fd3c44394d0>, {'action': u'login', 'controller': u'user'})
[Wed Aug 24 09:49:45.850861 2016] [wsgi:error] [pid 29401:tid 140547883652864] [remote 80.12.88.99:36581] ------------------------------------------------------------

**Comment by benscott** [comment]
Hi, I can't replicate this - it does seem to be working for me on CKAN 2.6 under Apache WSGI.  Could you post your ckanext.ldap.\* settings please and I'll try and dig into it some more.  Do you have any other CKAN plugins installed?

**Comment by gsueur** [comment]
Hi,

thanks for caring about this. Actually I've got a few more plugins installed, mostly data related : spatial_metadata spatial_query datapusher resource_proxy stats text_view image_view recline_view geojson_view pdf_view webpage_view datastore geo_view showcase dataproxy dataproxy_view
The LDAP is not accessible from the outside, but I succeed in connecting to it from a handmade python script. But maybe it was a misconfigured at the time I test the ldap plugin with paster serve. I'm gonna give it another try today, I'll tell you what happens. At least it's good to know that it _should_ work, I've been told the same on the ckan-dev list. 
Best regards
Guillaume
